src/main/proto/wfa/measurement/internal/kingdom/exchange_step_attempt_details.proto

```diff
@@ -0,0 +1,63 @@
+// Copyright 2021 The Cross-Media Measurement Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+syntax = "proto3";
+
+package wfa.measurement.internal.kingdom;
+
+import "google/protobuf/timestamp.proto";
+
+option java_package = "org.wfanet.measurement.internal.kingdom";
+option java_multiple_files = true;
+
+// Stores metadata about an ExchangeStepAttempt.
+message ExchangeStepAttemptDetails {
+  enum State {
+    STATE_UNSPECIFIED = 0;
+
+    // The attempt is has not yet reached a terminal state.
+    ACTIVE = 2;
+
+    // The attempt has succeeded. There cannot be more attempts for the step.
+    // Terminal state.
+    SUCCEEDED = 3;
+
+    // The attempt has failed. The step should be retried. Terminal state.
+    FAILED = 4;
+  }
+
+  // State of the ExchangeStepAttempt.
+  State state = 3;
+
+  message DebugLog {
```

Comment:
Where do you intend to write this?

Reply:
Can you clarify your question?

Reply:
Talked offline. Message will be written to the database.